### ISSUE TITLE
use-waitgroup-go: support for range statements

### ISSUE THREAD

**Issue by alexandear** [mgechev/revive#1685]
**Is your feature request related to a problem? Please describe.**

Currently, the rule `use-waitgroup-go:` does not report changing to `wg.Go(...)` in the following code:

```go
	wg := sync.WaitGroup{}

	wg.Add(int(jobs))

	for range int(jobs) {
		go func() {
			defer wg.Done()
			doSomething()
		}()
	}

	wg.Wait()
```

**Describe the solution you'd like**

Update the `use-waitgroup-go` rule.

**Additional context**

It's from https://github.com/kubernetes-sigs/kueue/pull/9892
